src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/DisableRuntimeMarshallingTests.cs

```diff
@@ -0,0 +1,1072 @@
+    {
+        {|CA1421:Marshal.PtrToStructure(ptr, (object)default(ValueType))|};
+        {|CA1421:Marshal.PtrToStructure(ptr, new ClassType())|};
+        _ = {|CA1421:Marshal.PtrToStructure(ptr, typeof(ManagedValueType))|};
+        _ = {|CA1421:Marshal.PtrToStructure(ptr, typeof(ValueType))|};
+        _ = {|CA1421:Marshal.PtrToStructure(ptr, typeof(ValueType?))|};
+        _ = {|CA1421:Marshal.PtrToStructure(ptr + 1, typeof(ValueType?))|};
+        _ = {|CA1421:Marshal.PtrToStructure(ptr, t)|};
+        _ = {|CA1421:Marshal.PtrToStructure<ValueType>(ptr)|};
+        _ = {|CA1421:Marshal.PtrToStructure<ValueType?>(ptr)|};
+        {|CA1421:Marshal.PtrToStructure<ValueType>(ptr, default(ValueType))|};
+        {|CA1421:Marshal.PtrToStructure<ClassType>(ptr, new ClassType())|};
+    }
+}
+
+struct ValueType
+{
+    int field;
+}
+struct ManagedValueType
+{
+    string field;
+}
+class ClassType
+{
+    int field;
+}
+";
+            string codeFix = @"
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+[assembly:DisableRuntimeMarshalling]
+
+class Foo
+{
+    public unsafe void Test(IntPtr ptr, Type t)
+    {
+        {|CA1421:Marshal.PtrToStructure(ptr, (object)default(ValueType))|};
+        {|CA1421:Marshal.PtrToStructure(ptr, new ClassType())|};
+        _ = {|CA1421:Marshal.PtrToStructure(ptr, typeof(ManagedValueType))|};
+        _ = (object)(*(ValueType*)ptr);
+        _ = (ValueType*)ptr is not null and var ptr1 ? *ptr1 : (object)null;
+        _ = (ValueType*)(ptr + 1) is not null and var ptr2 ? *ptr2 : (object)null;
+        _ = {|CA1421:Marshal.PtrToStructure(ptr, t)|};
+        _ = (*(ValueType*)ptr);
+        _ = (ValueType*)ptr is not null and var ptr3 ? *ptr3 : (ValueType?)null;
+        {|CA1421:Marshal.PtrToStructure<ValueType>(ptr, default(ValueType))|};
+        {|CA1421:Marshal.PtrToStructure<ClassType>(ptr, new ClassType())|};
+    }
+}
+
+struct ValueType
+{
+    int field;
+}
+struct ManagedValueType
+{
+    string field;
+}
+class ClassType
+{
+    int field;
+}
+";
+            await VerifyCSCodeFixAsync(source, codeFix, allowUnsafeBlocks: true);
+
+            // The code fix is not applicable when unsafe blocks are not allowed.
+            await VerifyCSCodeFixAsync(source, source, allowUnsafeBlocks: false);
+        }
+
+        private static Task VerifyCSAnalyzerAsync(string source, bool allowUnsafeBlocks = false)
+        {
+            return VerifyCSCodeFixAsync(source, source, allowUnsafeBlocks);
+        }
+
+        private static async Task VerifyCSCodeFixAsync(string source, string codeFix, bool allowUnsafeBlocks = false)
+        {
+            var test = new VerifyCS.Test
+            {
+                LanguageVersion = LanguageVersion.CSharp9,
+                ReferenceAssemblies = new ReferenceAssemblies("net7.0", new PackageIdentity("Microsoft.NETCore.App.Ref", "7.0.0-preview.1.22075.6"), Path.Combine("ref", "net7.0"))
```

Comment:
@sharwell Can this be added in the test library?

Reply:
Yes

Reply:
Though I'm hesitant to do so before 7.0.0-preview.1 is available on NuGet...

Reply:
Yeah I don't think we should update the testing SDK until we know for sure exactly which package version will be Preview 1. This just happened to be the newest Preview 1 package when I started working on this PR.